### PR TITLE
fix(mock): apply bare-key property overrides to nested objects (#3470)

### LINKED ISSUE #3470: Mock: `override.mock.properties` bare-key only matches top-level paths (not nested object properties like `PetWithTag.pet.name`)
### Description

`override.mock.properties` with a bare property-name key (e.g. `{ name: 'jon' }`) only substitutes the override when the property sits at the top level of the response schema. When the same-named property lives inside a non-array nested object (e.g. `PetWithTag.pet.name`), the override is silently dropped and the default per-type faker is emitted instead.

This is a sibling case to #2465 (which covers the array-of-objects case). The array case is being fixed in the corresponding PR; this issue tracks the same root mismatch for non-array nested objects, separated for a different blast-radius / backward-compatibility judgment.

### Output client

fetch (the matcher lives in the shared `@orval/mock` package, so all mock-emitting clients are affected — fetch, axios, msw, etc.)

### Configuration (orval.config)

```ts
export default defineConfig({
  petstore: {
    input: { target: 'spec/petstore.yaml' },
    output: {
      target: 'src/gen/petstore.ts',
      schemas: 'src/gen/model',
      client: 'fetch',
      mock: true,
      override: {
        mock: {
          properties: {
            name: 'jon',
          },
        },
      },
    },
  },
});
```

### Environment

- orval: 8.13.0 (current `master` HEAD)
- Node.js: 24.15.0
- OS: macOS 15

### Expected behavior

With `override.mock.properties = { name: 'jon' }`, every property literally named `name` produced by the mock generator — whether it's at the top of the response, inside an array of objects, or inside a nested object — should use the override value.

### Actual behavior

Only top-level `name` occurrences pick up the override. Nested-object occurrences fall back to the default per-type faker (`faker.string.alpha(...)`).

This is visible in the committed snapshot `tests/__snapshots__/mock/petstore/endpoints.ts`. With the petstore mock config setting `properties: { name: 'jon' }`:

| Operation | Returns | Snapshot `name` output | Notes |
| --- | --- | --- | --- |
| `createPets` / `showPetById` | `Pet` (top-level) | `name: 'jon'` ✅ | matches |
| `listPets` | `Pets` (array of `Pet`) | `name: faker.string.alpha(...)` ❌ | covered by #2465 (array case) |
| `showPetWithOwner` | `PetWithTag` where `pet` field is `Pet` | `name: faker.string.alpha(...)` ❌ | **this issue** (nested non-array object) |

### OpenAPI document (minimal)

```yaml
openapi: 3.0.3
info:
  title: Nested mock override repro
  version: 0.0.0
paths:
  /pet-with-owner:
    get:
      operationId: showPetWithOwner
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/PetWithTag'
components:
  schemas:
    Pet:
      type: object
      required: [id, name]
      properties:
        id:
          type: integer
        name:
          type: string
    PetWithTag:
      type: object
      required: [tag, pet]
      properties:
        tag:
          type: string
        pet:
          $ref: '#/components/schemas/Pet'
```

With `properties: { name: 'jon' }`, the generated mock emits `name: faker.string.alpha(...)` inside `pet`, not `name: 'jon'`.

#### Root cause (same family as #2465)

The matcher in `packages/mock/src/faker/resolvers/value.ts` (`resolveMockOverride`) does an exact-path equality:

```ts
if (`#.${key}` === path) return true;
```

The mock walker composes `path` as it descends:
- object property → append `.${key}` (`object.ts:184`)
- array items → append `.[]` (`scalar.ts:294`)

So for `PetWithTag.pet.name`, the path becomes `#.pet.name`, while a bare-key override `name` resolves to target `#.name` → no match.

The PR for #2465 narrows the fix to stripping `.[]` array-items markers only (so `#.[].name` → `#.name` → matches `name`). That fix intentionally **does not** touch the non-array nested case because:

1. #2465 specifically asks about arrays;
2. The `.[]` marker is internal-only (never documented as a key syntax);
3. Object nesting (`.pet.`) is meaningful structure that users currently use via explicit `#.pet.name` or regex `/^name$/` keys — extending bare-key transparency to nested objects is a behavior change with broader review surface and should be considered on its own.

This issue tracks whether to extend bare-key transparency to non-array nested objects too — see PR for #2465 for the related fix and decision rationale.

#### Possible fix sketch

Add an `item.name` direct-match fallback for bare keys in `resolveMockOverride`, gated only when the path **does not** include an explicit override target (so users who write `#.pet.name` to specifically target only that location keep current behavior). Or document `properties` as always matching by property name and require explicit-path / regex for scoping — this is a doc + small code change.

Either route needs:
- A regression spec covering nested objects (this issue's example),
- Snapshot updates in `tests/__snapshots__/mock/petstore/endpoints.ts` (the `PetWithTag.pet.name` line),
- A decision on whether the broader semantic change deserves a `feat!` or stays a `fix`.

### PR DESCRIPTION
**Pull request by wadakatu**
## Description

Bare-key `override.mock.properties` (e.g. `{ name: 'jon' }`) reached the response root and — since #2465 — array items, but stopped at **non-array nested objects**: `PetWithTag.pet.name` fell back to faker instead of the override.

This is the nested-object follow-up to #2465 / #3471, with the direction (Option A′) agreed by @melloware and @Hypenate in #3470.

### Before / After

petstore mock (`properties: { name: 'jon' }`):

| occurrence | path | before | after |
|---|---|---|---|
| top-level | `#.name` | `'jon'` | `'jon'` |
| array item (`listPets`) | `#.[].name` | `'jon'` (#2465) | `'jon'` |
| **nested object** (`PetWithTag.pet.name`) | `#.pet.name` | `faker.string.alpha(...)` | `'jon'` |

## How

`resolveMockOverride` now matches in two tiers:

1. **Tier 1 (unchanged)** — regex keys (against name or full path) and exact array-transparent path matching. Explicit keys (regex, or a dotted path like `country.name`) still win.
2. **Tier 2 (new)** — bare-name transparency: a dot-less key applies wherever the property literally appears, at any nesting depth, mirroring the array transparency from #2465. Only reached when no Tier 1 key matched, so **the most specific key always wins**.

This preserves the existing granular tools — dotted-path keys (`country.name`) and regex keys (`/^name$/`) behave exactly as before — so `country.name` vs `user.name` targeting is unaffected.

## Tests

- `packages/mock/src/faker/resolvers/value.test.ts`: the previously "still does not match nested" case is flipped to assert it now matches (#3470), plus a new describe block covering precedence (dotted key wins its path, bare key covers the rest) and anchoring (a dotted key stays root-anchored and does not bleed onto a deeper path).
- Snapshot: the only generated change across the whole suite is petstore `getShowPetWithOwnerResponseMock` nested `pet.name` flipping faker → `'jon'`.

### Local gates

- `@orval/mock` vitest: 229/229
- typecheck, lint: pass
- `tests` workspace build (typecheck-generated for 16 clients + verify-mock-generated): pass
- `test:snapshots`: 4582/4582 (1 snapshot updated)

Closes #3470

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved mock override matching with a two-tier precedence so specific path/regex overrides take priority while bare property-name overrides apply for nested non-array occurrences.

* **Tests**
  * Expanded tests covering nested override precedence and edge cases.
  * Updated mock snapshot to use a stable example value for an inner object name.

### CODE REVIEW

File: tests/__snapshots__/mock/petstore/endpoints.ts [hunk 1]
```diff
@@ -353,7 +353,7 @@ export const getShowPetWithOwnerResponseMock = (
     ]),
     '@id': faker.string.alpha({ length: { min: 10, max: 30 } }),
     id: faker.number.int({ min: 0, max: 100 }),
-    name: faker.string.alpha({ length: { min: 10, max: 30 } }),
+    name: 'jon',
```
[the remark this answers is not in the document]
Comment (wadakatu) [reply tests/__snapshots__/mock/petstore/endpoints.ts:356]: This file isn't hand-edited — it's regenerated output. `tests/__snapshots__/mock/petstore/endpoints.ts` is produced by `bun run test:snapshots:update`, which runs the real generator against `tests/configs/mock.config.ts` and writes the result here; `api-generation.spec.ts` then diffs freshly-generated output against it. That config sets `override.mock.properties: { name: 'jon' }`, so after this fix `name` is `'jon'` at every depth (including the nested `PetWithTag.pet.name` on line 356), while the other fields stay faker-random because the config only overrides `name`. The single-field change is exactly the intended generated diff — nothing was hardcoded.

File: packages/mock/src/faker/resolvers/value.ts [hunk 2]
```diff
@@ -59,6 +64,16 @@ export function resolveMockOverride(
...
 
+  // Tier 2 — bare property-name transparency (#3470): a dot-less key applies
+  // wherever the property literally appears, including inside non-array nested
+  // objects, mirroring the array transparency from #2465. Only reached when no
+  // explicit Tier 1 key matched, so it never overrides a more specific key.
+  if (!property) {
+    property = entries.find(
+      ([key]) => !isRegex(key) && !key.includes('.') && key === item.name,
+    );
+  }
```
[the remark this answers is not in the document]
Comment (wadakatu) [reply packages/mock/src/faker/resolvers/value.ts:75]: Intentional and agreed with the maintainers in #3470 (Option A′): @melloware and @Hypenate both signed off on bare keys matching nested objects, explicitly choosing it over a gated/opt-in option, and keeping it a `fix` since it only affects generated mock fixtures. Granularity isn't lost — explicit dotted-path keys (`country.name`) and regex keys (`/^name$/`) are unchanged and still take precedence (Tier 1). The changelog is generated automatically by release-please from the `fix(mock):` commit, and the existing docs already describe `properties` as "Match by property name (string or regex)", which this makes more consistent. A deeper write-up of nesting + precedence would be a nice docs follow-up but is out of scope for this fix.

File: packages/mock/src/faker/resolvers/value.ts [hunk 3]
```diff
@@ -44,7 +44,12 @@ export function resolveMockOverride(
   // Regex keys still match against the original (un-normalized) path so users
   // can opt into array-scoped targeting explicitly if ever needed.
   const normalizedPath = stripArrayMarkerSegments(path);
-  const property = Object.entries(properties).find(([key]) => {
+  const entries = Object.entries(properties);
```
[the remark this answers is not in the document]
Comment (wadakatu) [reply packages/mock/src/faker/resolvers/value.ts:47]: `properties` here is the user's `override.mock.properties` map — typically a handful of keys, not a hot-path collection — and the second `find` only runs in the no-match case. I kept the two explicit tiers deliberately: they make the precedence rule (regex/dotted-path wins over bare-name) obvious at a glance. A single-pass best-candidate tracker would trade that clarity for no measurable gain, so I'm leaving it as is.

File: packages/mock/src/faker/resolvers/value.ts [hunk 4]
```diff
@@ -44,7 +44,12 @@ export function resolveMockOverride(
...
+
+  // Tier 1 — explicit matches: regex (against name or full path) and exact
+  // array-transparent path. Checked first so a specific override (regex or
+  // dotted path like `country.name`) always wins over the bare-name fallback.
+  let property = entries.find(([key]) => {
```
[the remark this answers is not in the document]
Comment (wadakatu) [reply packages/mock/src/faker/resolvers/value.ts:52]: Same concern as the thread on line 47 — declining for the same reason (small user-config map, and the explicit tiers document the precedence). See that reply.

File: packages/mock/src/faker/resolvers/value.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (wadakatu) [reply packages/mock/src/faker/resolvers/value.ts:75]: Same concern as the thread on line 47 — declining for the same reason (small user-config map, and the explicit tiers document the precedence). See that reply.

**Review by melloware — APPROVED** [review]
